src/util/cache.ts

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright (c) 2020, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+export class Cache extends Map {
+  private static _instance: Cache;
+  private _hits = 0;
```

Comment:
wanna use the new stuff? `#hits = 0`

Reply:
Yeah, totally should. Typescript's `private` only treats methods as private during compile time, using `#` will do it both at compile and runtime.
https://www.amitmerchant.com/typescript-private-modifier-vs-ecmascript-hash-private-fields/